src/constraint_aware_append.c

```diff
@@ -360,13 +345,37 @@ constraint_aware_append_plan_create(PlannerInfo *root, RelOptInfo *rel, struct C
 	CustomScan *cscan = makeNode(CustomScan);
 	Plan *subplan = linitial(custom_plans);
 	RangeTblEntry *rte = planner_rt_fetch(rel->relid, root);
+	List *chunk_ris = NIL;
+	List *rt_index = NIL;
+	ListCell *lc_append;
 
 	cscan->scan.scanrelid = 0;			 /* Not a real relation we are scanning */
 	cscan->scan.plan.targetlist = tlist; /* Target list we expect as output */
 	cscan->custom_plans = custom_plans;
-	cscan->custom_private = list_make3(list_make1_oid(rte->relid),
-									   list_copy(root->append_rel_list),
-									   list_copy(clauses));
+
+	/*
+	 * create per chunk RestrictInfo
+	 *
+	 * We also need to walk the expression trees of the restriction clauses and
+	 * update any Vars that reference the main table to instead reference the child
+	 * table (chunk) we want to exclude.
+	 */
+	foreach (lc_append, root->append_rel_list)
```

Comment:
I think `root->append_rel_list` has the append relations for all append rels in the query and not just the `rel` you are processing. So I think this is incorrect. You need to include something like 
```
if (appinfo->parent_relid != relid)
				continue
```

Reply:
fixed